ecl/data_source_ecl_network_fic_gateway_v2.go

```diff
@@ -0,0 +1,126 @@
+package ecl
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/nttcom/eclcloud/ecl/network/v2/fic_gateways"
+)
+
+func dataSourceNetworkFICGatewayV2() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNetworkFICGatewayV2Read,
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_service_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"fic_gateway_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"qos_option_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"tenant_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceNetworkFICGatewayV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.networkV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL network client: %w", err)
+	}
+
+	var opts fic_gateways.ListOpts
+
+	if v, ok := d.GetOk("description"); ok {
+		opts.Description = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_service_id"); ok {
+		opts.FICServiceID = v.(string)
+	}
+
+	if v, ok := d.GetOk("fic_gateway_id"); ok {
+		opts.ID = v.(string)
+	}
+
+	if v, ok := d.GetOk("name"); ok {
+		opts.Name = v.(string)
+	}
+
+	if v, ok := d.GetOk("qos_option_id"); ok {
+		opts.QoSOptionID = v.(string)
+	}
+
+	if v, ok := d.GetOk("status"); ok {
+		opts.Status = v.(string)
+	}
+
+	if v, ok := d.GetOk("tenant_id"); ok {
+		opts.TenantID = v.(string)
+	}
+
+	pages, err := fic_gateways.List(client, opts).AllPages()
+	if err != nil {
+		return fmt.Errorf("unable to retrieve fic_gateways: %w", err)
+	}
+
+	gws, err := fic_gateways.ExtractFICGateways(pages)
+	if err != nil {
+		return fmt.Errorf("unable to extract fic_gateways: %w", err)
+	}
+
+	if len(gws) < 1 {
+		return fmt.Errorf("your query returned no results." +
+			" please change your search criteria and try again.")
```

Comment:
以下のように最後のドットを削除していただけますか？
```suggestion
			" please change your search criteria and try again")
```